components/uaepass-authenticator/src/main/java/org/wso2/carbon/identity/uaepass/authenticator/UAEPassAuthenticator.java

```diff
@@ -0,0 +1,1109 @@
+
+                if (StringUtils.isNotBlank(authenticatorProperties.get(UAEPassAuthenticatorConstants.UAE.QUERY_PARAMS))) {
+                    loginPage = processAdditionalQueryParamSeperation(authenticatorProperties, loginPage);
+                } else {
+                    Map<String, String> paramMap = new HashMap<>();
+                    paramMap.put(UAEPassAuthenticatorConstants.UAE.ACR_VALUES,
+                            UAEPassAuthenticatorConstants.UAEPassRuntimeConstants.DEFAULT_ACR_VALUES);
+                    paramMap.put(UAEPassAuthenticatorConstants.UAE.SCOPE,
+                            UAEPassAuthenticatorConstants.UAEPassRuntimeConstants.DEFAULT_SCOPES);
+                    loginPage = FrameworkUtils.buildURLWithQueryParams(loginPage, paramMap);
+                }
+
+                response.sendRedirect(loginPage);
+            } else {
+                throw new AuthenticationFailedException("Error while retrieving properties. "
+                        + "Authenticator Properties cannot be null");
+            }
+
+        } catch (IOException e) {
+            LOG.error("Authorization request building failed.");
+            throw new AuthenticationFailedException("Unable to pick correct env or problem occurred in additional " +
+                    "query params when building the authorize request", e);
+        } catch (OAuthSystemException e) {
+            LOG.error("Unable to build the request with must required query params.");
+            throw new AuthenticationFailedException("Exception while building authorization request with must " +
+                    "required query params", e);
+        }
+    }
+
+
+    /**
+     * Implements the logic of the UAE Pass federated authenticator.
+     *
+     * @param request
+     * @param response
+     * @param context
+     * @throws AuthenticationFailedException - exception while creating the access token or id token
+     */
+    @Override
+    protected void processAuthenticationResponse(HttpServletRequest request, HttpServletResponse response,
+                                                 AuthenticationContext context) throws AuthenticationFailedException {
+
+        try {
+            OAuthAuthzResponse authzResponse = OAuthAuthzResponse.oauthCodeAuthzResponse(request);
+            OAuthClientRequest accessTokenRequest = getAccessTokenRequest(context, authzResponse);
+            OAuthClient oAuthClient = new OAuthClient(new URLConnectionClient());
+            OAuthClientResponse oAuthResponse = getOauthResponse(oAuthClient, accessTokenRequest);
+
+            String accessToken = oAuthResponse.getParam(UAEPassAuthenticatorConstants.UAE.ACCESS_TOKEN);
+            if (StringUtils.isBlank(accessToken)) {
+                if(LOG.isDebugEnabled()){
+                    LOG.debug("Access Token is empty");
+                }
+                throw new AuthenticationFailedException("Access token is empty");
```

Comment:
```suggestion
                throw new AuthenticationFailedException("Access token is empty.");
```

Check other places as well